libraries/chain/steem_evaluator.cpp

```diff
@@ -600,7 +600,14 @@ void comment_evaluator::do_apply( const comment_operation& o )
             FC_ASSERT( _db.calculate_discussion_payout_time( *parent ) != fc::time_point_sec::maximum(), "Discussion is frozen." );
       }
 
-      if( _db.has_hardfork( STEEM_HARDFORK_0_12__176 ) )
+      if( _db.has_hardfork( STEEM_HARDFORK_0_20__2019 ) )
+      {
+         if( o.parent_author == STEEM_ROOT_POST_PARENT )
+             FC_ASSERT( ( now - auth.last_root_post ) > STEEM_MIN_ROOT_COMMENT_INTERVAL, "You may only post once every 5 minutes.", ("now",now)("last_root_post", auth.last_root_post) );
+         else
+             FC_ASSERT( (now - auth.last_post) > STEEM_MIN_REPLY_INTERVAL_HF20, "You may only comment once every 3 seconds.", ("now",now)("auth.last_post",auth.last_post) );
```

Comment:
A similar comparison is being done here to allow one vote every three seconds:
https://github.com/steemit/steem/blob/bb8ca476f33c08434274f2d76333fcb74cfa58b8/libraries/chain/steem_evaluator.cpp#L1229

With the vote compassion, it it is using `>=` instead of `>`. Would this need to be changed to `>=` in order to allow one comment per block (every 3 seconds)?

Reply:
You have a very good point, and I guess the comparison could be changed this way more to be consistent with the vote comparison, but from a functional point of view `(now - auth.last_post) > STEEM_MIN_REPLY_INTERVAL_HF20` only allows one comment to be made per block, because it is something like `time since last comment must be strictly greater than the block time`.

I will update the PR as you guys think, after all, you know best!